beacon-chain/slasher/receive_test.go

```diff
@@ -14,6 +14,204 @@ import (
 	logTest "github.com/sirupsen/logrus/hooks/test"
 )
 
+func Test_processQueuedAttestations(t *testing.T) {
+	type args struct {
+		attestationQueue []*slashertypes.CompactAttestation
+		currentEpoch     types.Epoch
+	}
+	tests := []struct {
+		name                 string
+		args                 args
+		shouldNotBeSlashable bool
+		wantedLogs           []string
+	}{
+		{
+			name: "Detects surrounding vote (source 1, target 2), (source 0, target 3)",
+			args: args{
+				attestationQueue: []*slashertypes.CompactAttestation{
+					{
+						AttestingIndices: []uint64{0, 1},
+						Source:           1,
+						Target:           2,
+					},
+					{
+						AttestingIndices: []uint64{0, 1},
+						Source:           0,
+						Target:           3,
+					},
+				},
+				currentEpoch: 4,
+			},
+			wantedLogs: []string{"Attester surrounding vote"},
+		},
+		{
+			name: "Detects surrounding vote (source 50, target 51), (source 0, target 1000)",
+			args: args{
+				attestationQueue: []*slashertypes.CompactAttestation{
+					{
+						AttestingIndices: []uint64{0, 1},
+						Source:           50,
+						Target:           51,
+					},
+					{
+						AttestingIndices: []uint64{0, 1},
+						Source:           0,
+						Target:           1000,
+					},
+				},
+				currentEpoch: 4,
+			},
+			wantedLogs: []string{"Attester surrounding vote"},
+		},
+		{
+			name: "Detects surrounded vote (source 0, target 3), (source 1, target 2)",
+			args: args{
+				attestationQueue: []*slashertypes.CompactAttestation{
+					{
+						AttestingIndices: []uint64{0, 1},
+						Source:           0,
+						Target:           3,
+					},
+					{
+						AttestingIndices: []uint64{0, 1},
+						Source:           1,
+						Target:           2,
+					},
+				},
+				currentEpoch: 4,
+			},
+			wantedLogs: []string{"Attester surrounded vote"},
+		},
+		{
+			name: "Not slashable, surrounding but non-overlapping attesting indices",
+			args: args{
+				attestationQueue: []*slashertypes.CompactAttestation{
+					{
+						AttestingIndices: []uint64{0, 1},
+						Source:           1,
+						Target:           2,
+					},
+					{
+						AttestingIndices: []uint64{2, 3},
+						Source:           0,
+						Target:           3,
+					},
+				},
+				currentEpoch: 4,
+			},
+			shouldNotBeSlashable: true,
+		},
+		{
+			name: "Not slashable, (source 1, target 2), (source 2, target 3)",
+			args: args{
+				attestationQueue: []*slashertypes.CompactAttestation{
+					{
+						AttestingIndices: []uint64{0, 1},
+						Source:           1,
+						Target:           2,
+					},
+					{
+						AttestingIndices: []uint64{0, 1},
+						Source:           2,
+						Target:           3,
+					},
+				},
+				currentEpoch: 4,
+			},
+			shouldNotBeSlashable: true,
+		},
+		{
+			name: "Not slashable, (source 1, target 2), (source 1, target 2)",
```

Comment:
I don't feel 100% comfortable seeing a double vote pass. Will this test case change after double vote slashing is implemented?